src/cmd/flash.rs

```diff
@@ -152,18 +152,19 @@ pub fn run(args: &Args) -> Result<()> {
                 "flashing local image other than `--version latest` is not yet supported"
             ));
         }
-        let variant = if args.recovery { "recovery" } else { "test" };
+        let variant = if args.recovery { "signed" } else { "test" };
         format!("xBuddy://{host}/{board_to_flash}/{version}/{variant}")
     };
 
     // Determine a destination
-    let destination = match (&args.dut, args.usb) {
-        (Some(dut), false) => {
+    let destination = match (&args.dut, args.usb, args.recovery) {
+        (Some(dut), false, false) => {
             ensure_testing_rsa_is_there()?;
             let dut = &DutInfo::new(dut)?;
             dut.ssh().host_and_port()
         }
-        (None, true) => "usb://".to_string(),
+        (Some(_), false, true) => bail!("Please specify --usb instead of DUT for --recovery"),
```

Comment:
nit: maybe "Recovery image is not for flashing via SSH. Please specify --usb as a destination instead of --dut." or something would be more helpful for the users.

Reply:
OK, let me update it.